src/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -52,7 +52,7 @@
     private BinaryCodec mStream = null;
 
     // Underlying compressed data stream.
-    private final BlockCompressedInputStream mCompressedInputStream;
+    private BlockCompressedInputStream mCompressedInputStream;
```

Comment:
I don't love making this non-final and doing null checking everywhere.  It's probably reasonable to think that upon `BAMFileReader.close()` the reader itself is released and ready for GC soon thereafter, such that there's not much to gain from releasing the `BlockCompressedInputStream` a little faster.